tests/PHPStan/Analyser/data/array-flip.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace ArrayFlip;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @param int[] $integerList
+ */
+function foo($integerList)
+{
+	$flip = array_flip($integerList);
+	assertType('array<int, (int|string)>', $flip);
+}
+
+/**
+ * @param mixed[] $list
+ */
+function foo3($list)
+{
+	$flip = array_flip($list);
+
+	if (PHP_VERSION_ID < 80000) {
+		assertType('array|null', $flip);
+	} else {
+		assertType('array', $flip);
+	}
```

Comment:
it seems this is not the right way to expect different types based on php version

Reply:
It isn't, you need to `yield from` different files and do that `if` in the data provider.